app/models/course.rb

```diff
@@ -382,23 +382,33 @@ def next_recruitment_cycle?
   # Ideally this would just use the validation, but:
   # https://github.com/rails/rails/issues/13971
   def course_params_assignable(course_params)
-    entry_requirements_assignable(course_params) &&
+    assignable_after_publish(course_params) &&
+      entry_requirements_assignable(course_params) &&
       qualification_assignable(course_params)
   end
 
 private
 
+  def assignable_after_publish(course_params)
```

Comment:
I find this massive chain of methods pretty hard to follow what's happening, wonder if there's a way to split it up into a few intermediate variables/methods